### ISSUE TITLE
과제 2에서 코드 관련 질문 있습니다. 

### ISSUE THREAD

**Issue by soheekim911** [heartcored98/Standalone-DeepLearning#8]
안녕하세요 수업 잘 듣고 있습니다. 
과제를 하다가 궁금한 점이 있었는데 이제야 질문드립니다..!

1. cifar10 dataset에서 데이터를 확인하기 위한 부분에서 나온 코드입니다. 
이 떄 img = img/ 2+0.5 부분이 unnormalize하는 것이라고 설명을 써두셨는데요, 
애초에 데이터셋에서는 왜 저렇게 어둡게 normalize를 해둔 것일까요?
 `def imshow(img): 
    img = img / 2 + 0.5     # unnormalize`

2. 이 부분은 experiment define하는 셀 중 train 부분인데, 5번째 줄에서 optimizer.zero_grad()가 나와서  
epoch마다 gradient를 리셋하는 것은 알겠는데 맨 끝줄 training 중에는 왜 optimizer.zero_grad()가 나오는지 모르겠습니다.

`   
       
        # ==== Train ===== #
        net.train()
        optimizer.zero_grad()
        
        running_loss = 0.0
        train_loss = 0.0

        for i, data in enumerate(trainloader, 0):
            # get the inputs
            inputs, labels = data
            inputs = inputs.view(-1, 3072)
            
            inputs = inputs.cuda()
            labels = labels.cuda()
            

            # zero the parameter gradients
            optimizer.zero_grad() ###
`

**Comment by soheekim911** [comment]
이해가 되었습니다. 답변 감사드립니다!!
